### ISSUE TITLE
oc downloaded on each start of Minishift

### ISSUE THREAD

**Issue by glennodickson** [minishift/minishift#206]
**Minishift  1.0.0-beta.1 + Windows + VirtualBox**

Hi 
I'm starting and stopping minishift on window 10 using the following OC commands:
```
minishift start
...
...
minishift stop
```

Every start up I see a download of 18.42MB and everything is reset again and basically loose all my projects. If this is usual/expected behaviour then what is the oc delete used for in that case?

`6.52 MB / 18.42 MB [======================================================================>---------------------------------------------------------------------------------------------------------------------------------]  35.41% 1m37`

**Comment by LalatenduMohanty** [comment]
@glennodickson Are you using HyperV?

**Comment by glennodickson** [comment]
Hi Thanks for your reply, no using VirtualBox latest update.

**Comment by hferentschik** [comment]
> Every start up I see a download of 18.42MB and everything is reset again and basically loose all my projects. If this is usual/expected behaviour then what is the oc delete used for in that case?

No, this is not the expected behavior. I would assume that the 18MB is the download of the _oc_ binary which is then used to provision OpenShift. This binary should get cached under _~/.minishift/cache/oc/v1.3.1/oc~. It would help to have some more context. Could you run minishift like this:

    $ minishift -v 2 --show-libmachine-logs start --vm-driver virtualbox

And also check and report the contents of your _~/.minishift_ directory.

Regarding your OpenShift config. Try the following:

    $ minishift start --vm-driver virtualbox --use-existing-config true

ATM it is expected that your OpenShift config disappears when you "just" start/stop the VM. This is how 'oc cluster up' works. You need to explicitly set the _use-existing-config_. This needs to be documented. In the long run, minishift will take care of this and on restart automatically specify this flag.

**Comment by hferentschik** [comment]
See also issue #209 and #210

**Comment by glennodickson** [comment]
Hi. Thanks for your reply. 

I hope I've managed to provide you with all the information you might need below. 

**Step1: I checked the following directory state before starting minishift:**

Here is the state of the directories before execution:
**Before Start** 

_\.minishift contains:_
![image](https://cloud.githubusercontent.com/assets/3872513/21393120/fd97d35a-c7cd-11e6-9693-1939ecbaeb6a.png)

_.minishift\cache\oc\v1.3.1 - contains oc.exe_
![image](https://cloud.githubusercontent.com/assets/3872513/21393039/a7a685cc-c7cd-11e6-89e6-cb431277f04a.png)

_\.minishift\cache\iso contains:_
![image](https://cloud.githubusercontent.com/assets/3872513/21393070/c80b0004-c7cd-11e6-9bc9-7ef175bfa760.png)

_\.minishift\cache\openshift is **empty**._

_\.minishift\machines\minishift contains:_
![image](https://cloud.githubusercontent.com/assets/3872513/21392980/6178a396-c7cd-11e6-96b2-b2299b6890cc.png)

**Step 2: Executed Start and Results:**
I started and stopped Minishift to see if it would retain the below security changes between restarting. Executed the following commands:

```
C:\Users\dickson>minishift -v 2 --show-libmachine-logs start --vm-driver virtualbox
Starting local OpenShift cluster...
Waiting for SSH to be available...
Detecting the provisioner...
Waiting for SSH to be available...
Detecting the provisioner...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...

This machine has been allocated an IP address, but Docker Machine could not
reach it successfully.

SSH for the machine should still work, but connecting to exposed ports, such as
the Docker daemon port (usually <ip>:2376), may not work properly.

You may need to add the route manually, or use another related workaround.

This could be due to a VPN, proxy, or host file configuration issue.

You also might want to clear any VirtualBox host only interfaces you are not using.
 18.42 MB / 18.42 MB [==========================================================================================================] 100.00% 0s
Provisioning OpenShift via 'C:\Users\dickson\.minishift\cache\oc\v1.3.1\oc [cluster up]'
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ...
   Deleted existing OpenShift container
-- Checking for openshift/origin:v1.3.1 image ... OK
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ... OK
-- Checking type of volume mount ...
   Using Docker shared volumes for OpenShift volumes
-- Creating host directories ... OK
-- Finding server IP ...
   Using 192.168.99.100 as the server IP
-- Starting OpenShift container ...
   Creating initial OpenShift configuration
   Starting OpenShift using container 'origin'
   Waiting for API server to start listening
   OpenShift server started
-- Installing registry ... OK
-- Installing router ... OK
-- Importing image streams ... OK
-- Importing templates ... OK
-- Login to server ... OK
-- Creating initial project "myproject" ... OK
-- Server Information ...
   OpenShift server started.
   The server is accessible via web console at:
       https://192.168.99.100:8443

   You are logged in as:
       User:     developer
       Password: developer

   To login as administrator:
       oc login -u system:admin

C:\Users\dickson>oc login -u system:admin
Logged into "https://192.168.99.100:8443" as "system:admin" using existing credentials.

You have access to the following projects and can switch between them with 'oc project <projectname>':
    default
    kube-system
  * myproject
    openshift
    openshift-infra

Using project "myproject".

C:\Users\dickson>oc adm policy add-cluster-role-to-user cluster-admin developer

C:\Users\dickson>oc adm policy add-cluster-role-to-user cluster-admin admin
```

**Step 3 : Confirmed that both users can see all projects**
Yes they both can.

**Step 4: Stopped and Restarted Minishift**
```
minishift stop 
minishift -v 2 --show-libmachine-logs start --vm-driver virtualbox
```
**Step 5: Checked to see if admin or developer user could see all projects.**
_Results_ : Neither could see all projects

**Directories After Restart** 
_.minishift contains:_
![image](https://cloud.githubusercontent.com/assets/3872513/21393655/5f1d7088-c7d0-11e6-8534-8ee9212abbc6.png)

_.minishift\cache\oc\v1.3.1 - contains oc.exe - datetime changes - **maybe since its downloaded again**_
![image](https://cloud.githubusercontent.com/assets/3872513/21393764/c181bca2-c7d0-11e6-8fd6-ba559f02ae6e.png)

_C:\Users\dickson\.minishift\cache\iso - no change at all._
![image](https://cloud.githubusercontent.com/assets/3872513/21393792/f2f1fb3a-c7d0-11e6-9cd2-b840fecc41c2.png)

_C:\Users\dickson\.minishift\cache\openshift is empty. **Is this right?**_

_\.minishift\machines\minishift contains:_
![image](https://cloud.githubusercontent.com/assets/3872513/21393724/9879f36a-c7d0-11e6-9c6d-2208fb0c9f84.png)

I executed the second statement you mentioned in your previous email with the same results.
Hope this enough to solve this. 

Thanks

**Comment by hferentschik** [comment]
@glennodickson, awesome thanks for the detailed context. Helps a lot. Your descriptions seems to hint to the fact that the algorithm to check whether the oc binary exists does not work on Windows and hence it gets re-downloaded every time, effectively overwriting it. I need to check the code and also try this myself on Windows to see whether I can re-produce this.

**Comment by hferentschik** [comment]
Resolved via pull request #253
